src/type_object.rs

```diff
@@ -31,17 +58,14 @@ pub trait PyTypeInfo {
     /// Class doc string
     const DESCRIPTION: &'static str = "\0";
 
-    /// Size of the rust PyObject structure (PyObject + rust structure)
-    const SIZE: usize;
-
-    /// `Type` instance offset inside PyObject structure
-    const OFFSET: isize;
-
     /// Type flags (ie PY_TYPE_FLAG_GC, PY_TYPE_FLAG_WEAKREF)
     const FLAGS: usize = 0;
 
     /// Base class
-    type BaseType: PyTypeInfo;
+    type BaseType: PyTypeInfo + PyTypeObject;
+
+    /// Layout
+    type ConcreteLayout: PyConcreteObject<Self>;
```

Comment:
Is it necessary for the `ConcreteLayout` to be part of the `TypeInfo` trait? It seems like this is only needed for `#[pyclass]` types. Having it here creates a lot of work to get the right definitions for all the py native types, but I'm not sure if it helps.

Reply:
If we want to give a fixed size to `PyClassShell`, we need it.

> Having it here creates a lot of work to get the right definitions for all the py native types,

Yeah but I'll do it by script.